service/kas/access/rewrap.go

```diff
@@ -141,16 +167,55 @@ func verifySignedRequestToken(ctx context.Context, in *kaspb.RewrapRequest) (*Re
 	switch publicKey := clientPublicKey.(type) {
 	case *rsa.PublicKey:
 		requestBody.PublicKey = publicKey
-		return requestBody, nil
+		return &requestBody, nil
 	case *ecdsa.PublicKey:
 		requestBody.PublicKey = publicKey
-		return requestBody, nil
+		return &requestBody, nil
 	default:
 		slog.WarnContext(ctx, fmt.Sprintf("clientPublicKey not a supported key, was [%T]", clientPublicKey))
 		return nil, err400("clientPublicKey unsupported type")
 	}
 }
 
+func verifySRT(ctx context.Context, srt string, dpopJWK jwk.Key) (string, error) {
+	v, err := jws.NewVerifier(jwa.RS256)
```

Comment:
Do we want to fix the algorithm here? `RS256` is pretty popular but it seems like it might be good to allow others.

Reply:
ah, interesting, I thought the hash size was part of the name of the algorithm. but yeah, we control this so there isn't really a point to supporting more things.